test/SerialIOTest.cpp

```diff
@@ -1660,32 +1682,65 @@ void fileBased_write_test(const std::string & backend)
         e["positionOffset"]["x"].makeConstant(1.23);
 
         fileBased_add_EDpic(e, 42);
+        it.setTime(static_cast< double >(10));
 
-        REQUIRE(o.iterations.size() == 1);
+        REQUIRE(o.iterations.size() == 2);
     }
     REQUIRE((auxiliary::file_exists("../samples/subdir/serial_fileBased_write10." + backend)
         || auxiliary::directory_exists("../samples/subdir/serial_fileBased_write10." + backend)));
 
     // read back with auto-detection and non-fixed padding
     {
         Series s = Series("../samples/subdir/serial_fileBased_write%T." + backend, Access::READ_ONLY);
-        REQUIRE(s.iterations.size() == 7);
+        REQUIRE(s.iterations.size() == 8);
+        REQUIRE(s.iterations.contains(4));
+        REQUIRE(s.iterations.contains(10));
+        REQUIRE(s.iterations.contains(123456));
+
+        REQUIRE(s.iterations[3].time< double >() == 3.0);
+        REQUIRE(s.iterations[4].time< double >() == 4.0);
+        REQUIRE(s.iterations[5].time< double >() == 5.0);
+        REQUIRE(s.iterations[10].time< double >() == 10.0);
+        REQUIRE(s.iterations[123456].time< double >() == double(123456));
     }
 
-    // write with auto-detection and in-consistent padding
+    // write with auto-detection and in-consistent padding from step 10
     {
         REQUIRE_THROWS_WITH(Series("../samples/subdir/serial_fileBased_write%T." + backend, Access::READ_WRITE),
             Catch::Equals("Cannot write to a series with inconsistent iteration padding. Please specify '%0<N>T' or open as read-only."));
     }
 
-    // read back with auto-detection and fixed padding
+    // read back with fixed padding
+    {
+        Series s = Series("../samples/subdir/serial_fileBased_write%03T." + backend, Access::READ_ONLY);
+        REQUIRE(s.iterations.size() == 7);
+        REQUIRE(s.iterations.contains(4));
+        REQUIRE(!s.iterations.contains(10));
+        REQUIRE(s.iterations.contains(123456));
+
+        REQUIRE(s.iterations[3].time< double >() == 3.0);
+        REQUIRE(s.iterations[4].time< double >() == 4.0);
+        REQUIRE(s.iterations[5].time< double >() == 5.0);
+    }
+
+    // read back with auto-detection (allow relaxed/overflow padding)
     {
-        Series s = Series("../samples/subdir/serial_fileBased_write%08T." + backend, Access::READ_ONLY);
-        REQUIRE(s.iterations.size() == 6);
+        Series s = Series("../samples/subdir/serial_fileBased_write%T." + backend, Access::READ_ONLY);
+        REQUIRE(s.iterations.size() == 8);
+        REQUIRE(s.iterations.contains(4));
+        REQUIRE(s.iterations.contains(10));
+        REQUIRE(s.iterations.contains(123456));
+
+        REQUIRE(s.iterations[3].time< double >() == 3.0);
+        REQUIRE(s.iterations[4].time< double >() == 4.0);
+        REQUIRE(s.iterations[5].time< double >() == 5.0);
+        REQUIRE(s.iterations[10].time< double >() == 10.0);
+        REQUIRE(s.iterations[123456].time< double >() ==
+            static_cast<double>(123456));
```

Comment:
Style consistency in the block:
```suggestion
            static_cast< double >(123456));
```